src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
```

Comment:
```suggestion
    private static final int DEFAULT_CAPACITY = 10;
```

Reply:
you don't need a separate global variable for storing current array's length, you can get current length using method of array itself